src/components/v5/common/ActionSidebar/partials/motions/MotionSimplePayment/steps/VotingStep/utils.tsx

```diff
@@ -0,0 +1,53 @@
+import { MotionVote } from '~utils/colonyMotions';
+import { formatText } from '~utils/intl';
+
+export const renderVoteRadioButtons = (
+  hasUserVoted: boolean,
+  userVote: number,
+) => {
+  const supportOption = {
+    label: formatText({ id: 'motion.support' }),
+    id: 'support',
+    value: MotionVote.Yay,
+    colorClassName: 'text-purple-200',
+    iconName: 'thumbs-up',
+  };
+  const opposeOption = {
+    label: formatText({ id: 'motion.oppose' }),
+    id: 'oppose',
+    value: MotionVote.Nay,
+    colorClassName: 'text-negative-300',
+    iconName: 'thumbs-down',
+  };
+
+  if (!hasUserVoted) {
+    return [opposeOption, supportOption];
+  }
+
+  if (userVote === MotionVote.Nay) {
+    return [supportOption];
+  }
+
+  return [opposeOption];
+};
+
+export const setLocalStorageVoteValue = (
+  transactionId: string,
+  vote: number,
+) => {
+  const previousVotes = JSON.parse(localStorage.getItem('votes') || '{}');
```

Comment:
I think you should keep this data in separate keys. The key should consist of txId and some constant. Like `${txId}-vote`